Update PR template and the PR contributing guidelines (#1320)
OpenVisualCloud/SVT-AV1: CONTRIBUTING.md

## Contribution process

- Follow the [coding guidelines](STYLE.md)
- Validate that your changes do not break a build
  - either locally or through travis-ci and appveyor. Preferably all of them.
- Perform smoke tests and ensure they pass
- Submit a pull request for review to the maintainer

## Pull request process:

> - Make commits concise and to the point (1 feature / issue = 1 commit )

Reviewer: ```suggestion
- Make clear and concise commits (1 commit per 1 feature or issue)
```

Author: applied

> - Authors would be responsible for breaking down the PR into commits that make sense (with proper commit messages)

Reviewer: ```suggestion
- The author will be responsible for breaking down the PR into sensible commits with proper commit messages
```

Author: applied

> - Use rebase and merge for all PRs to make sure that all commits are carried over to the master

Reviewer: ```suggestion
- Use rebase to make sure all commits can apply cleanly onto the master branch
```

`Rebase and merge` is mostly a github UI thing

Author: applied

> - Maintainers would only use squash and merge with the permission of the authors.

Reviewer 2: Maybe add some guideline about `force push`? If it is just to address comments, no need to use force push. And please choose `yes` for squash and merge, so that maintainer can handle it.

> - Maintainers would only use squash and merge with the permission of the authors.

Reviewer: ```suggestion
- Maintainers shall only use `squash and merge` with the permission of the authors.
```

Small wording change that makes it more like an actual guideline. Plus it makes it sound like it's not just for the PR author, but also for the maintainers too.

Author: applied
